runtime/js/fetch.js

```diff
@@ -0,0 +1,66 @@
+import { fetch as fetchImpl } from "ext:deno_fetch/26_fetch.js";
+import { fromInnerResponse, toInnerResponse } from "ext:deno_fetch/23_response.js";
+import { toInnerRequest, fromInnerRequest, Request } from "ext:deno_fetch/23_request.js";
+import { guardFromHeaders } from "ext:deno_fetch/20_headers.js";
+
+let ipfsScheme = "ipfs://";
+let ipfsBaseUrl = undefined;
+
+export function setLassieUrl(/** @type {string} */ value) {
+  ipfsBaseUrl = value + "ipfs/";
+}
+
+export function fetch(resource, options) {
+  let request = new Request(resource, options);
+  // Fortunately, Request#url is a string, not an instance of URL class
```

Comment:
What is fortunate about that?

Reply:
I decided to write a longer comment, see 7db4ee8

```js
  // The `resource` arg can be a string or any other object with a stringifier — including a URL
  // object — that provides the URL of the resource you want to fetch; or a Request object.
  // See https://developer.mozilla.org/en-US/docs/Web/API/fetch#parameters
  // Fortunately, Request's constructor handles the conversions, and Request#url is always a string.
  // See https://developer.mozilla.org/en-US/docs/Web/API/Request/url
```